### PR TITLE
New shield code

### PR DESCRIPTION
**Pull request by Taranchuk**

### CODE REVIEW

**Comment by Taranchuk** [comment]
New changes:

Shields are now counted as apparel instead of equipment. This is done so that we can draw them in the apparel class, instead of in the pavn renderer transpiler. If possible, additional patches can be made to virtually place them in the equipment section.

Removed stats worker patch. As it turned out, it does not affect armor stats at all, it handles in a separate patch. Still not fully understood the purpose of this patch, but the combat is still unchanged after the removal of this stats worker patch.

There are changes in the medieval mod, we need to put this xml block:
```
		<thingClass>VFECore.Apparel_Shield</thingClass>
		<apparel>
			<bodyPartGroups>
				<li>LeftHand</li>
			</bodyPartGroups>
			<layers>
				<li>VFEC_OuterShell</li>
			</layers>
		</apparel>
```
into <ThingDef Name="VFEM_BaseShield"

**Comment by Taranchuk** [comment]
Otherwise everything is unchanged, no functionality has been removed or altered

File: Source/VFECore/HarmonyPatches/HarmonyPatches.cs [hunk 1]
```diff
@@ -44,11 +44,11 @@ static HarmonyPatches()
             // Facial Stuff
             if (ModCompatibilityCheck.FacialStuff)
             {
-                var humanBipedDrawer = GenTypes.GetTypeInAnyAssembly("FacialStuff.HumanBipedDrawer", "FacialStuff");
-                if (humanBipedDrawer != null)
-                    VFECore.harmonyInstance.Patch(AccessTools.Method(humanBipedDrawer, "DrawApparel"), transpiler: new HarmonyMethod(typeof(Patch_PawnRenderer.RenderPawnInternal), "Transpiler"));
-                else
-                    Log.Error("Could not find type FacialStuff.HumanBipedDrawer in Facial Stuff");
+                //var humanBipedDrawer = GenTypes.GetTypeInAnyAssembly("FacialStuff.HumanBipedDrawer", "FacialStuff");
```
Comment (erdelf) [inline Source/VFECore/HarmonyPatches/HarmonyPatches.cs:47]: this whole section is not required anymore, due to moving out of the pawn renderer I take it. Remove if so.

File: Source/VFECore/Utilities/ShieldUtility.cs [hunk 2]
```diff
@@ -83,7 +83,7 @@ public static void MakeRoomForShield(this Pawn_EquipmentTracker equipment, Thing
             
         }
 
-        public static void AddShield(this Pawn_EquipmentTracker equipment, ThingWithComps newEq)
+        public static void AddShield(this Pawn_EquipmentTracker equipment, ThingWithComps newEq, Pawn wearer)
```
Comment (erdelf) [inline Source/VFECore/Utilities/ShieldUtility.cs:86]: Is there any advantage to giving both equipment tracker and wearer, given the usage seems to imply the equipment tracker is of the wearer ?

File: Source/VFECore/Apparels/Apparel_Shield.cs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                return true;
+            }
+            Lord lord = this.Wearer.GetLord();
+            if (lord != null && lord.LordJob != null && lord.LordJob.AlwaysShowWeapon)
+            {
+                return true;
+            }
+            return false;
+        }
+        private Vector3 GetAimingVector(Vector3 rootLoc)
```
Comment (erdelf) [inline Source/VFECore/Apparels/Apparel_Shield.cs:38]: a whole lot of magic numbers in this method. Would benefit from a few comments.

**Review by erdelf — CHANGES_REQUESTED** [review]
Code in general needs a partial rewrite, but for conversion to apparel seems good.
(Didn't manually check if the drawloc checks out)

File: Source/VFECore/Utilities/ShieldUtility.cs (hunk 2 above)
Comment (Taranchuk) [reply Source/VFECore/Utilities/ShieldUtility.cs]: i changed the method in new commits so it now uses the wearer

File: Source/VFECore/Apparels/Apparel_Shield.cs (hunk 3 above)
Comment (Taranchuk) [reply Source/VFECore/Apparels/Apparel_Shield.cs:38]: these numbers are from vanilla DrawEquipment method

File: Source/VFECore/HarmonyPatches/HarmonyPatches.cs (hunk 1 above)
Comment (Taranchuk) [reply Source/VFECore/HarmonyPatches/HarmonyPatches.cs:48]: done

**Comment by lbmaian** [comment]
> Removed stats worker patch. As it turned out, it does not affect armor stats at all, it handles in a separate patch. Still not fully understood the purpose of this patch, but the combat is still unchanged after the removal of this stats worker patch.

@Taranchuk It's for adding stat modifiers for non-shield/armor-related stats like a move speed modifier, e.g.

```xml
		<equippedStatOffsets>
			<ShootingAccuracyPawn>-0.1</ShootingAccuracyPawn>
			<MoveSpeed>-0.5</MoveSpeed>
		</equippedStatOffsets>
```

With your change to make shields part of apparel, this isn't needed anymore, since stat modifiers from worn apparel are automatically considered in StatsWorker. The patch was only necessary for equipment because StatsWorker only looked at the primary equipment.

I've also considered making a similar change in JecsTools (making shields apparel and/or removing StatsWorker patch), which shares ChJeez Pawn Shields heritage with this mod, for performance reasons, but thought it would be too much of a breaking change for legacy framework, especially since some JecsTools shields are also weapons.

**Pull request merged**